var/spack/repos/builtin/packages/gmp/package.py

```diff
@@ -37,6 +37,9 @@ class Gmp(AutotoolsPackage, GNUMirrorPackage):
     )
     variant("cxx", default=True, description="Enable C++ support")
 
+    # can be removed upon  >= v6.2.2
```

Comment:
Can remove this comment and replace with a URL to the patch source. Could even download the patch directly, might be better to avoid licensing issues.

Reply:
Also, did you confirm the patch applies correctly to all older versions?

Reply:
> Can remove this comment and replace with a URL to the patch source. Could even download the patch directly, might be better to avoid licensing issues.

I'd looked at this but the url for the patch (https://gmplib.org/repo/gmp/raw-rev/5f32dbc41afc) doesn't end in `.diff` and the patch function doesn't seem to respect an `extension="diff"` kwarg passed to it.

> Also, did you confirm the patch applies correctly to all older versions?
I hadn't! Good catch! It looks like this patch only works for v6.2.1. I'll amend the recipe as such.

Reply:
I don't think an extension is required, is it?

Reply:
The fetcher seems to fail without it, although maybe I'm doing something wrong? Do you also get an all fetchers failed error replacing the patch with the one below?

```python
patch(
    "https://gmplib.org/repo/gmp/raw-rev/5f32dbc41afc",
    sha256="a44ef57903b240df6fde6c9d2fe40063f785995c43b6bfc7a237c571f53613e0",
    when="@:6.2.1",
)
```

Reply:
Yeah it doesn't work for me either, no idea why.